magnum_cluster_api/resources.py

```diff
@@ -271,6 +271,88 @@ def get_object(self) -> objects.ClusterResourceSet:
         )
 
 
+class StorageClassesConfigMap(ClusterBase):
+    def __init__(
+        self,
+        context: context.RequestContext,
+        api: pykube.HTTPClient,
+        cluster: magnum_objects.Cluster,
+    ):
+        self.context = context
+        self.api = api
+        self.cluster = cluster
+
+    def get_object(self) -> pykube.ConfigMap:
+        osc = clients.get_openstack_api(self.context)
+        volume_types = osc.cinder().volume_types.list()
+        data = {}
+        for vt in volume_types:
+            data[f"{vt.name}.yaml"] = yaml.dump(
+                {
+                    "apiVersion": objects.StorageClass.version,
+                    "kind": objects.StorageClass.kind,
+                    "metadata": {
+                        "annotations": {
+                            "storageclass.kubernetes.io/is-default-class": "true"
+                        }
+                        if vt.name == "__DEFAULT__"
+                        else {},
+                        "name": vt.name,
+                    },
+                    "provisioner": "kubernetes.io/cinder",
+                    "parameters": {
+                        "type": vt.name,
+                    },
+                    "reclaimPolicy": "Delete",
+                    "volumeBindingMode": "Immediate",
+                }
+            )
+        return pykube.ConfigMap(
+            self.api,
+            {
+                "apiVersion": pykube.ConfigMap.version,
+                "kind": pykube.ConfigMap.kind,
+                "metadata": {
+                    "name": "storageclasses",
+                    "namespace": "magnum-system",
+                },
+                "data": data,
+            },
+        )
+
+
+class StorageClassesClusterResourceSet(ClusterBase):
+    def get_object(self) -> objects.ClusterResourceSet:
+        version = utils.get_cluster_label(
+            self.cluster, "cinder_csi_plugin_tag", CSI_TAG
+        )
+
+        return objects.ClusterResourceSet(
+            self.api,
+            {
+                "apiVersion": objects.ClusterResourceSet.version,
+                "kind": objects.ClusterResourceSet.kind,
+                "metadata": {
+                    "name": "storageclasses",
```

Comment:
```suggestion
                    "name": "cinder-csi-storageclass",
```

I'm thinking we should do this to avoid clashes